main.go

```diff
@@ -0,0 +1,14 @@
+package main
+
+import (
+	"github.com/gorilla/mux"
+	"log"
+	"net/http"
+)
+
+func main() {
+	myRouter := mux.NewRouter().StrictSlash(true)
+	deviceService := deviceHTTPHandler{service: deviceService{dao: &inMemoryDeviceDAO{}}}
+	myRouter.HandleFunc("/devices", deviceService.createDevice).Methods("POST")
+	log.Fatal(http.ListenAndServe(":8000", myRouter))
```

Comment:
Port should be configurable using environment variable. You can set the default in code but you should be prepared to accept different port.